iCount/files/fastq.py

```diff
@@ -1,213 +1,59 @@
-# pylint: skip-file
 """.. Line to protect from pydocstyle D205, D400.
 
 FASTQ
 -----
 
 Reading and writting `FASTQ`_ files.
 """
-import gzip
-import math
 import os
 
+import iCount
 
-# precompute, so mapping is faster
-_dX2L = [x - 64 for x in range(59, 105)]
-_dX2L = [10.0 * math.log(math.pow(10.0, x / 10.0) + 1.0, 10.0) for x in _dX2L]
-_dX2L = [chr(math.floor(x) + 33) for x in _dX2L]
 
-_Dx2l = {chr(ij): v for ij, v in zip(range(59, 105), _dX2L)}
+class FastqEntry:
+    """Single FASTQ entry."""
 
+    def __init__(self, id, seq, plus, qual):  # pylint: disable=redefined-builtin
+        """Initialize attributes."""
+        self.id = str(id).split(' ')[0]  # pylint: disable=invalid-name
+        self.seq = str(seq)
+        self.plus = str(plus)
+        self.qual = str(qual)
 
-def _x2l(s):
-    """TODO."""
-    # Q_solexa = [ord(x) - 64 for x in s]
-    # Q_PHRED = [10.0 * math.log(math.pow(10.0, x / 10.0) + 1.0, 10.0) for x in Q_solexa]
-    # L = [chr(math.floor(x) + 33) for x in Q_PHRED]
-    # L = [_dX2L[ord(x) - 59] for x in x]
-    # return ''.join(L)
-    return ''.join((_Dx2l[ij] for ij in s))
+    def __repr__(self):
+        """Represent object."""
+        return self.id
 
 
-# -31 = -64 + 33
-_Dij2l = {chr(ij): chr(ij - 31) for ij in range(64, 104 + 1 + 1)}
+class FastqFile:
+    """Write FASTQ files."""
 
+    def __init__(self, fname, mode='rt'):
+        """Open file handle in desired mode."""
+        if 'r' in mode and not os.path.isfile(fname):
+            self.file = None
+            raise FileNotFoundError('File not found.')
 
-def _ij2l(s):
-    """TODO."""
-    return ''.join((_Dij2l[ij] for ij in s))
-
-
-# 0 = -33 + 33
-_Dsl2l = {chr(ij): chr(ij - 0) for ij in range(64, 104 + 1 + 1)}
-
-
-def _sl2l(s):
-    """TODO."""
-    # return ''.join((_Dsl2l[ij] for ij in s))
-    return s
-
-
-def _determine_format(fname):
-    """
```

Comment:
Can you please split this PR into three separate:
- "Restructure fastq module" and "Fix if xlsite is on begging of chrom" into "fastq and xlsites: minor fixes"
- "Restructure fastq module" into "fastq: restructure", should include the `_determine_format` functionallity.
- "Refactor demultiplex" into "demultiplex: refactoring"

(please try to follow the formatting of PR titles, i.e., module: description)